test/directives/omp_simd_clause.f90

```diff
@@ -13,9 +13,8 @@ subroutine sum(myarr1,myarr2,ub)
   end do
 end subroutine
 
-! CHECK-NOT:  {{.*}} add nsw <[[VF:[0-9]+]] x i32>{{.*}}
-! METADATA-NOT: load {{.*}}, !llvm.mem.parallel_loop_access ![[TAG1:[0-9]+]]
-! METADATA-NOT: store {{.*}}, !llvm.mem.parallel_loop_access ![[TAG1]]
-! METADATA-NOT: ![[TAG2:[0-9]+]] = !{!"llvm.loop.vectorize.enable", i1 true}
-! METADATA-NOT: ![[TAG1:[0-9]+]] = distinct !{![[TAG1]], ![[TAG2]]}
+! CHECK-NOT:  {{.*}} add nsw <{{[0-9]+}} x i32>{{.*}}
+! METADATA-NOT: llvm.mem.parallel_loop_access
+! METADATA-NOT: llvm.loop.vectorize.enable
+! METADATA-NOT: ![[TAG1:[0-9]+]] = distinct !{![[TAG1]], !{{[0-9]+}}}
```

Comment:
If I remember correctly, this type of self-referential metadata is common for other purposes. If we no longer link the second tag to the `llvm.loop.vectorize.enable` metadata, then we probably don't need this FileCheck pattern.

Why do we need to relax these patterns anyway? What IR does the new compiler generate for these test cases?

Reply:
The IR is actually the same, I don't understand why I got this error on release_13x:
```
/home/ARM/flang/test/directives/omp_simd_clause.f90:18:65: error: undefined variable: TAG1
! METADATA-NOT: store {{.*}}, !llvm.mem.parallel_loop_access ![[TAG1]]
                                                                ^
/home/ARM/flang/test/directives/omp_simd_clause.f90:20:61: error: undefined variable: TAG2
! METADATA-NOT: ![[TAG1:[0-9]+]] = distinct !{![[TAG1]], ![[TAG2]]}
                                                            ^

Input file: <stdin>
Check file: /home/ARM/flang/test/directives/omp_simd_clause.f90

-dump-input=help explains the following input dump.

Input was:
<<<<<<
          1: ; ModuleID = '/tmp/omp_simd_clause-023d2a.ll'
not:18'0     X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: match failed for invalid pattern
not:18'1                                                    undefined variable: TAG1
not:20'0     X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: match failed for invalid pattern
not:20'1                                                    undefined variable: TAG2
```
I don't know how much time should I dig into this issue since the changes does work for both releases

Reply:
What are the full IRs produced for this `RUN` command by the two branch? The differences between the IRs should explain why the FileCheck patterns fail to match.

Reply:
The differences are not significant:
```
--- a/flang-omp.rel12.txt
+++ b/flang-omp.rel13.txt
@@ -1,9 +1,9 @@
-; ModuleID = '/tmp/omp_simd_clause-e50b97.ll'
-source_filename = "/tmp/omp_simd_clause-e50b97.ll"
+; ModuleID = '/tmp/omp_simd_clause-ff7991.ll
+source_filename = "/tmp/omp_simd_clause-ff7991.ll"
 target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
 target triple = "x86_64-unknown-linux-gnu"

-; Function Attrs: nofree norecurse nounwind
+; Function Attrs: nofree norecurse nosync nounwind
 define void @sum_(i64* nocapture readonly %"myarr1$p", i64* nocapture readonly %"myarr2$p", i64* nocapture readonly %ub, i64* nocapture readonly %"myarr1$sd", i64* nocapture readonly %"myarr2$sd") local_unnamed_addr #0 !dbg !5 {
 L.entry:
   %0 = bitcast i64* %ub to i32*, !dbg !16
@@ -58,7 +58,7 @@ L.LB1_312:                                        ; preds = %L.LB1_311, %L.entry
   ret void, !dbg !20
 }

-attributes #0 = { nofree norecurse nounwind }
+attributes #0 = { nofree norecurse nosync nounwind }

 !llvm.module.flags = !{!0, !1}
 !llvm.dbg.cu = !{!2}

```

Reply:
I see, that makes sense! Then the change is fine, although as I said in my first comment, we probably should delete the third `METADATA-NOT` pattern as well.

Reply:
done, removed third `METADATA-NOT` as requested